### PR TITLE
[feature] add vscode workspace config for linting + debugging

### PR DESCRIPTION
**Pull request by f0x52**
Having this in the repository will allow any dev using vscode (or forks etc) to automatically get the correct linter and debug run target set up.

### CODE REVIEW

**Comment by tsmethurst** [comment]
Maybe it's better to put these in the CONTRIBUTING.md as copy-pasteable snippets under the setting up your dev environment section. I'm always a bit wary about adding editor-specific things in this way. What do you think? I could be being ridiculous :upside_down_face:

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{
+	"version": "0.2.0",
```
[the remark this answers is not in the document]
Comment (f0x52) [reply .vscode/launch.json:2]: it doesn't launch on start, but it adds a launch target under the 'run and debug' sidebar menu, so you can start the testrig with a single click + have it attached to vscode's debugger / error tracing etc

**Comment by f0x52** [comment]
Yeah I think CONTRIBUTING.md would be fine too, I'll change the PR to add it there + add the `.vscode` folder to .gitignore then?

**Comment by tsmethurst** [comment]
Yep perfect! ❤️❤️

Sent from ProtonMail mobile

-------- Original Message --------

**Comment by igalic** [comment]
I honestly think it's fine to commit whatever the main developers are using, and the farther the spread, the easier it's for new contributors to get started by just cloning the repo

**Pull request merged**